### PR TITLE
Red thread

### PR DESCRIPTION
**Pull request by Ivikhostrup**
# What was implemented
The report was read through with the goal of checking that head and tail was there for each section.

# Explanation
.

# Outcome
The report should have better coherency

### CODE REVIEW

File: report/sections/Chapter 01 - Introduction/Intro.tex [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 This resulted in a survey being sent out, and a student task force being assembled to understand the issues that caused the high failure rate.
-Here, many students mentioned not feeling experienced enough with the programming languages used in the course to teach the functional and logical programming paradigms.
+Here, many students mentioned not feeling experienced enough with the programming languages used in the course to teach the functional and logical programming paradigms. \todo{Insert citation}
 
 
 Consequently, \aau{} chose to restructure the course entirely.
 Rather than teaching many different programming languages within the functional and logical programming paradigms, the course was restructured to focus only on Haskell.
 Furthermore, the course now features more practical programming exercises with classroom discussion in-between, rather than traditional lectures followed by a separate exercise session.
-Supporting this way of teaching, the course lectures have been moved to a more classical classroom setup.
+Supporting this way of teaching, the course lectures have been moved to a more classical classroom setup. \todo{Could be worded differently}
```
Comment (TheRumle) [inline report/sections/Chapter 01 - Introduction/Intro.tex:18]:
```suggestion
To accommodate the course changes, the course lecture have been changed into a classroom setup where lecturing and exercises are interweaved.
```

File: report/sections/Chapter 01 - Introduction/Intro.tex [hunk 2]
```diff
@@ -38,7 +38,7 @@ \section{Existing solutions}
 The feedback loop for each platform is quite different, however.
 
 
-Codewars focuses on competitive programming, meaning users compete to solve problems fastest and most efficient way.
+Codewars focuses on competitive programming, meaning users compete to solve problems fastest and most efficient way. \todo{Spelling}
```
Comment (TheRumle) [inline report/sections/Chapter 01 - Introduction/Intro.tex:41]:
```suggestion
Codewars focuses on competitive programming where users compete to solve problems.
```

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 3]
```diff
@@ -1,19 +1,17 @@
-The following chapter contains short introductions to the most important technologies used in the project. 
+The following chapter contains short introductions to the most important technologies used in the project. Additionally, we also introduce relevant theory.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:1]:
```suggestion
The following chapter contains short introductions to the most important technologies used in the project. Additionally, theory necessary to understand the various components of the developed platform is presented.
```

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 4]
```diff
@@ -34,13 +32,13 @@ \section*{Rust}
 Rust is a multi-paradigm, general-purpose programming language originally developed by Graydon Hoare in 2006 as a personal project at Mozilla. 
 The project was then sponsored by Mozilla from 2009 to 2021 where the project along with all trademarks was moved to the newly create Rust Foundation. 
 The idea behind Rust was to challenge the idea that high-level ergonomics and low-level control come at the cost of each other\cite{Rust_Book}.
-Most other programming languages today operate in a band where either safety or control is valued highest with the other diminished. Such as C and its high level of control and lack of safety. 
+Most other programming languages today operate in a band where either safety or control is valued highest and the other diminished. An example of this is C that has high level control but lacks safety.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:35]:
```suggestion
Most other programming languages today operate in a band where either safety or control is valued highest and the other diminished. An example of this is the C Language that offers high level of control but lacks safety. \todo{source}
```

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 5]
```diff
@@ -34,13 +32,13 @@ \section*{Rust}
...
 
-In order to achieve this goal Rust provides a large number of safety features among these is its most distinguishing feature which prevents invalid data access at compile time. 
-Considering a research article by Microsoft Security Research Center stating that invalid data access is behind 70\% of serious security bugs in modern programs this is a large deal\cite{Safe_Systems_Languages}. 
-Instead of relying on the developer to deal with it, Rust's compiler guarantees any compiled program is memory-safe.
+In order to achieve this goal Rust provides a large number of safety features, among these is its most distinguishing feature which prevents invalid data access at compile time. 
+Considering a research article by Microsoft Security Research Center stating that invalid data access is behind 70\% of serious security bugs in modern programs, the benefits that Rust provide are significant\cite{Safe_Systems_Languages}. 
+Instead of relying on the developer to manage memory, Rust's compiler guarantees that any compiled program is memory-safe.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:41]: I would like to have a souce on guarantee of memory-safety.

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 6]
```diff
@@ -51,16 +49,16 @@ \section*{Rust}
 \section*{Docker}
 The main purpose of Docker is to simplify deliveries in distributed systems with the use of so-called containers\cite{Docker_Container}.
 Containers are bundles of code and required dependencies. This allows developers to move around prepackaged applications ensuring that the developed software runs the same regardless of what system is used\cite{Docker_Container}.
-The containers themselves are created from an image. Here the image is the executable package that encapsulates everything required in order to run the application. On runtime, the image is then turned into our container. 
+The containers themselves are created from an image. Here the image is the executable package that encapsulates everything required in order to run the application. On runtime, the image is then turned into a container. 
 This is possible due to the Linux kernel's build process isolation and virtualization capabilities. These also allow a single host system to share its resources with multiple application components\cite{Docker_Container}.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:53]: "These also allow a single"   er "these" containers eller capabilities?

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 7]
```diff
@@ -51,16 +49,16 @@ \section*{Rust}
...
 
-We use Docker to set up local development environments for the APIs and Databases. This allows for an easier development process and a simplified delivering process to production.
+We use Docker to set up local development environments for the APIs and Databases. This allows for an easier development process and a simplified delivery process to production.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:55]: Do we dockerize all production?

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 8]
```diff
@@ -51,16 +49,16 @@ \section*{Rust}
...
 
 \section*{tRPC}
 tRPC is a library that enables developers to build type safe APIs without having to deal with schemas or code generation. 
 It is structured in a way that allows for types to be shared between the client and server by only importing the types instead of the actual server code. 
 This in turn allows developers to not have the actual server code exposed in the \frontend{}. 
-Furthermore, tRPC is created in a way to fully take advantage of the power of a full-stack TypeScript project, which a language-agnostic framework\cite{tRPC}.
+Furthermore, tRPC is created in a way to fully take advantage of the power of a full-stack TypeScript project, which is a language-agnostic framework\cite{tRPC}.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:61]: How does it take advantage? Does "which" refer to tRPC or TS?

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 9]
```diff
@@ -74,15 +72,15 @@ \section*{Prisma}
 \section*{TypeScript}
 As the server-side use of \javascript{} keeps growing in popularity its inability to provide developers with access to strong type checking and compile time error checking continues to prevent it from being the go-to language.
 As a way to address these problems, \typescript{} was designed to try and bridge this gap without requiring developers to learn a new language.
-Instead, \typescript{} was developed as a syntactic superset to \javascript{} allowing developers to keep building into their already existing knowledge but with the addition of additional features. 
+Instead, \typescript{} was developed as a syntactic superset to \javascript{} allowing developers to keep building into their already existing knowledge but with additional features.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:75]:
```suggestion
Instead, \typescript{} was developed as a syntactic superset to \javascript{} allowing developers to extend existing knowledge and features with typesafe functionality.
```

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 10]
```diff
@@ -74,15 +72,15 @@ \section*{Prisma}
 \section*{TypeScript}
 As the server-side use of \javascript{} keeps growing in popularity its inability to provide developers with access to strong type checking and compile time error checking continues to prevent it from being the go-to language.
 As a way to address these problems, \typescript{} was designed to try and bridge this gap without requiring developers to learn a new language.
-Instead, \typescript{} was developed as a syntactic superset to \javascript{} allowing developers to keep building into their already existing knowledge but with the addition of additional features. 
+Instead, \typescript{} was developed as a syntactic superset to \javascript{} allowing developers to keep building into their already existing knowledge but with additional features. 
 This also allows for all \javascript{} projects to be converted to \typescript{} projects allowing for easier movement between the two languages\cite{TypeScript}.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:76]:
```suggestion
This means that all \javascript{} projects are syntactically valid in typescript. This in turn allows a development team to gradually adopt typescript into their project and use \typescript{} and \javascript{} interchangeably. \cite{TypeScript}. 
```

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 11]
```diff
@@ -74,15 +72,15 @@ \section*{Prisma}
...
 This is possible because \typescript{} itself is compiled into \javascript{}, which is an interpreted language. 
 
 \section*{Next.js}
-Next.js (Next) is an open-source React framework created by Vercel that enables react web applications to use server-side rendering as well as generate static websites. The Next framework was created in order to counter the potential drawbacks of React relying on client-side rendering such as potential security issues. By enabling websites to be fully or partially server-side rendered instead Next provides a solution to this issues\cite {Nextjs_Docks}. 
+Next.js (Next) is an open-source React framework created by Vercel that enables react web applications to use server-side rendering as well as generate static websites. The Next framework was created in order to counter the potential security issues of React, which relies on client-side rendering. Next provides a solution to this by enabling websites to be fully or partially server-side rendered\cite {Nextjs_Docks}.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:81]:
```suggestion
Next.js (Next) is an open-source React framework created by Vercel that enables react web applications to use server-side rendering as well as generate static websites. Next was created to counter the potential security issues of React, which relies on client-side rendering. Next provides a solution  by enabling websites to be fully or partially server-side rendered\cite {Nextjs_Docks}. 
```

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex [hunk 12]
```diff
@@ -74,15 +72,15 @@ \section*{Prisma}
...
+Next.js (Next) is an open-source React framework created by Vercel that enables react web applications to use server-side rendering as well as generate static websites. The Next framework was created in order to counter the potential security issues of React, which relies on client-side rendering. Next provides a solution to this by enabling websites to be fully or partially server-side rendered\cite {Nextjs_Docks}. 
 
-Next as a framework also tries to fix another issue of React, namely that creating a website from the ground up still requires a lot of effort spending time configuring tools as well as rebuilding solutions. 
+Next as a framework also tries to fix another issue of React, namely that creating a website from the ground up still requires a lot of effort, such as spending time on configuring tools as well as rebuilding solutions.
```
Comment (TheRumle) [inline report/sections/Chapter 02 - Preliminaries/preliminaries.tex:83]:
```suggestion
Next  tries to fix another issue of React, namely that creating a website from the ground up still requires a lot of effort, such as spending time on configuring tools as well as rebuilding solutions. 
```

**Review by TheRumle — CHANGES_REQUESTED** [review]

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex (hunk 12 above)
Comment (chhoumann) [reply report/sections/Chapter 02 - Preliminaries/preliminaries.tex:83]: We should probably refer to it as Next.js everywhere, as that's how the docs also refer to it.

BTW, double space between Next and tries.

File: report/sections/Chapter 02 - Preliminaries/preliminaries.tex (hunk 11 above)
Comment (chhoumann) [reply report/sections/Chapter 02 - Preliminaries/preliminaries.tex:81]: We should probably refer to it as Next.js everywhere, as that's how the docs also refer to it.

**Pull request merged**